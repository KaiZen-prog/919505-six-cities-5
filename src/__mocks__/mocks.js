import {Cities, SortingTypes} from "../const";

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const InitialState = {
  APP_DATA: {
    offers: [],
    favoriteOffers: [],
    isFavoriteOffersLoaded: false,
    offerDetails: {},
    isOfferDetailsLoaded: false,
    nearbyOffers: [],
    isNearbyOffersLoaded: false,
    reviews: [],
    isReviewRequestPosted: false,
    postReviewError: null
  },

  APP_PROCESS: {
    currentCity: Cities[0],
    isOffersSortOpened: false,
    currentOffersSort: SortingTypes.POPULAR,
    activeCard: null,
    clickedCard: null
  },

  USER: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    data: null
  }
};

export const offers = [{
  city: `Cologne`,
  cityCoords: [50.938361, 6.959974],
  mapZoom: 13,
  coords: [50.957361, 6.9509739999999995],
  detailsMapZoom: 16,
  title: `The house among olive `,
  rating: 3.8,
  type: `house`,
  maxAdults: 3,
  price: 493,
  id: 1,
  bedroomsQuantity: 3,
  features: [
    `Laptop friendly workspace`,
    `Washer`,
    `Towels`,
    `Fridge`,
    `Air conditioning`,
    `Baby seat`,
    `Breakfast`
  ],
  poster: `"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"`,
  photos: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
  ],
  owner: {
    avatar: `img/avatar-angelina.jpg`,
    name: `Angelina`,
    isPro: true
  },
  isPremium: false,
  isInBookmarks: false,
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
},
{
  city: `Brussels`,
  cityCoords: [50.846557, 4.351697],
  mapZoom: 13,
  coords: [50.839557, 4.346697],
  detailsMapZoom: 16,
  title: `Amazing and Extremely Central Flat`,
  rating: 4.3,
  type: `apartment`,
  maxAdults: 3,
  price: 323,
  id: 2,
  bedroomsQuantity: 3,
  features: [
    `Towels`,
    `Baby seat`,
    `Laptop friendly workspace`,
    `Dishwasher`,
    `Washing machine`,
    `Air conditioning`,
    `Washer`,
    `Coffee machine`,
    `Breakfast`,
    `Fridge`
  ],
  poster: `"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"`,
  photos: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`
  ],
  owner: {
    avatar: `img/avatar-angelina.jpg`,
    name: `Angelina`,
    isPro: true
  },
  isPremium: false,
  isInBookmarks: false,
  description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
},
{
  city: `Brussels`,
  cityCoords: [50.846557, 4.351697],
  mapZoom: 13,
  coords: [50.865556999999995, 4.371696999999999],
  detailsMapZoom: 16,
  title: `The house among olive `,
  rating: 4.6,
  type: `apartment`,
  maxAdults: 4,
  price: 218,
  id: 3,
  bedroomsQuantity: 4,
  features: [
    `Washer`,
    `Laptop friendly workspace`,
    `Breakfast`,
    `Air conditioning`,
    `Baby seat`
  ],
  poster: `"https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"`,
  photos: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
  ],
  owner: {
    avatar: `img/avatar-angelina.jpg`,
    name: `Angelina`,
    isPro: true
  },
  isPremium: false,
  isInBookmarks: false,
  description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
}
];

export const offersFromServer = [{"city": {"name": `Cologne`, "location": {"latitude": 50.938361, "longitude": 6.959974, "zoom": 13}}, "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`], "title": `The house among olive `, "is_favorite": false, "is_premium": false, "rating": 3.8, "type": `house`, "bedrooms": 3, "max_adults": 4, "price": 493, "goods": [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`, `Breakfast`], "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`}, "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`, "location": {"latitude": 50.957361, "longitude": 6.9509739999999995, "zoom": 16}, "id": 1}, {"city": {"name": `Brussels`, "location": {"latitude": 50.846557, "longitude": 4.351697, "zoom": 13}}, "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`], "title": `Amazing and Extremely Central Flat`, "is_favorite": false, "is_premium": false, "rating": 4.3, "type": `apartment`, "bedrooms": 3, "max_adults": 8, "price": 323, "goods": [`Towels`, `Baby seat`, `Laptop friendly workspace`, `Dishwasher`, `Washing machine`, `Air conditioning`, `Washer`, `Coffee machine`, `Breakfast`, `Fridge`], "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`}, "description": `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`, "location": {"latitude": 50.839557, "longitude": 4.346697, "zoom": 16}, "id": 2}, {"city": {"name": `Brussels`, "location": {"latitude": 50.846557, "longitude": 4.351697, "zoom": 13}}, "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`], "title": `The house among olive `, "is_favorite": false, "is_premium": false, "rating": 4.6, "type": `apartment`, "bedrooms": 4, "max_adults": 7, "price": 218, "goods": [`Washer`, `Laptop friendly workspace`, `Breakfast`, `Air conditioning`, `Baby seat`], "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`}, "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`, "location": {"latitude": 50.865556999999995, "longitude": 4.371696999999999, "zoom": 16}, "id": 3}];

export const reviewsFromServer = [{"id": 1, "user": {"id": 1, "is_pro": false, "name": `xyz`, "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`}, "rating": 4, "comment": `12311231123112311231123112311231123112311231123112311231`, "date": `2020-11-18T15:41:58.083Z`}, {"id": 2, "user": {"id": 1, "is_pro": false, "name": `xyz`, "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`}, "rating": 3, "comment": `52525252525252525252525252525252525252525252525252`, "date": `2020-11-19T07:00:10.385Z`}];

export const userData = {
  avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
  email: `qwe@qwe.com`,
  id: 1,
  name: `Some User`,
};

export const infoFromServer = {"id": 1, "email": `xyz@xyz.com`, "name": `xyz`, "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`, "is_pro": false};

export const reviews = [{
  id: 1,
  author: {
    avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
    name: `Paul`,
    isPro: false,
  },
  rating: 4,
  text: `Nice`,
  date: `2020-11-15T13:58:19.494Z`,
},

{
  id: 2,
  author: {
    avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/4.jpg`,
    name: `John`,
    isPro: true,
  },
  rating: 3,
  text: `Not too good`,
  date: `2020-10-15T13:58:19.494Z`,
}
];


export const adaptReviewToApp = (comment) => {
  return {
    id: comment.id,
    author: {
      avatar: comment.user.avatar_url,
      name: comment.user.name,
      isPro: comment.user.is_pro
    },
    rating: comment.rating,
    text: comment.comment,
    date: comment.date,
  };
};


export const user = {
  "email": `Oliver.conner@gmail.com`,
  "password": `12345678`
};
