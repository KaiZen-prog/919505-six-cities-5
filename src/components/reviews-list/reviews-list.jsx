import React, {useEffect} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import ReviewsItem from "../rewiews-item/reviews-item";
import {fetchReviewsList} from "../../store/api-actions";

const ReviewsList = (props) => {
  const {offerId, reviews, getReviewsAction} = props;

  useEffect(() => {
    getReviewsAction(offerId);
  }, []);

  return (
    <>
      <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
      <ul className="reviews__list">
        {reviews.map((review, i) => (
          <ReviewsItem
            review={review}
            key={i}
          />
        ))}
      </ul>
    </>
  );
};

ReviewsList.propTypes = {
  offerId: PropTypes.number.isRequired,
  reviews: PropTypes.array.isRequired,
  getReviewsAction: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  getReviewsAction(offerId) {
    dispatch(fetchReviewsList(offerId));
  },
});

const mapStateToProps = (state) => {
  return {
    offerId: state.APP_PROCESS.clickedCard,
    reviews: state.APP_DATA.reviews,
  };
};


export {ReviewsList};
export default connect(mapStateToProps, mapDispatchToProps)(ReviewsList);
