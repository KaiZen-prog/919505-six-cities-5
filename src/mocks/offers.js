import {nanoid} from "nanoid";
import moment from "moment";

import {
  getRandomArrayElement,
  getRandomBool,
  getRandomInteger,
  createRandomArray
} from "../utils/common.js";

const OFFERS_QUANTITY = 4;

const TITLES = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
  `Wood and stone place`
];

const USERS = [
  {
    avatar: `avatar-angelina.jpg`,
    name: `Angelina`
  },

  {
    avatar: `avatar-max.jpg`,
    name: `Max`
  }
];

const POSTERS = [
  `apartment-01.jpg`,
  `apartment-02.jpg`,
  `apartment-03.jpg`,
  `studio-01.jpg`,
  `room.jpg`
];

const APARTMENT_TYPES = [
  `Apartment`,
  `Private room`
];

const PRICES = {
  MIN: 10,
  MAX: 500
};

const RATING_SCALE = {
  MIN: 0,
  MAX: 50
};

const BEDROOMS = {
  MIN: 1,
  MAX: 5
};

const MAX_ADULTS = {
  MIN: 1,
  MAX: 5
};

const FEATURES = [
  `Wi-Fi`,
  `Washing machine`,
  `Towels`,
  `Heating`,
  `Coffee machine`,
  `Baby seat`,
  `Kitchen`,
  `Dishwasher`,
  `Cable TV`,
  `Fridge`,
];

const DESCRIPTION = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra. Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante. Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui. Sed sed nisi sed augue convallis suscipit in sed felis. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`;

const generateReview = () => ({
  author: getRandomArrayElement(USERS),
  rating: getRandomInteger(RATING_SCALE.MIN, RATING_SCALE.MAX) / 10,
  text: DESCRIPTION.split(`.`, getRandomInteger(1, 5)).toString(),
  date: moment(Date.now() - getRandomInteger(0, 5 * 12 * 30 * 24 * 60 * 60 * 1000)).format(`YYYY-MM-DD`),
});

const getReviews = (num) => Array.from({length: num}, generateReview);

const generateOffer = () => ({
  id: nanoid(),
  title: getRandomArrayElement(TITLES),
  poster: getRandomArrayElement(POSTERS),
  photos: createRandomArray(POSTERS),
  type: getRandomArrayElement(APARTMENT_TYPES),
  price: getRandomInteger(PRICES.MIN, PRICES.MAX),
  rating: getRandomInteger(RATING_SCALE.MIN, RATING_SCALE.MAX) / 10,
  bedroomsQuantity: getRandomInteger(BEDROOMS.MIN, BEDROOMS.MAX),
  maxAdults: getRandomInteger(MAX_ADULTS.MIN, MAX_ADULTS.MAX),
  features: createRandomArray(FEATURES),
  description: DESCRIPTION.split(`.`, getRandomInteger(1, 5)).toString(),
  isPremium: getRandomBool(),
  owner: getRandomArrayElement(USERS),
  reviews: getReviews(getRandomInteger(0, 5)),
  isInBookmarks: getRandomBool()
});

const getOffers = () => {
  const offers = [];
  for (let i = 0; i < OFFERS_QUANTITY; i++) {
    let newOffer = generateOffer();
    offers.push(newOffer);
  }

  return offers;
};

export default getOffers;
