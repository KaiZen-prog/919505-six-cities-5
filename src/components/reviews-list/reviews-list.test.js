import React from "react";
import renderer from "react-test-renderer";
import {ReviewsList} from "./reviews-list";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import {offers, reviews} from "../../__mocks__/mocks";

const mockStore = configureStore();

const initialState = {
  APP_DATA: {
    reviews
  }
};

const store = mockStore(initialState);
const noop = () => {};

test(`ReviewsList without error`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter history={browserHistory}>
            <ReviewsList
              offerId={offers[0].id}
              reviews={reviews}
              getReviewsAction={noop}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

