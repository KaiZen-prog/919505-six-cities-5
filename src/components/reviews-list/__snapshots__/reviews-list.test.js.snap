// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList without error 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="ReviewsList avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Paul
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Nice
        </p>
        <time
          className="reviews__time"
          dateTime="2020-11-15T13:58:19.494Z"
        >
          November 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="ReviewsList avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/4.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          John
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Not too good
        </p>
        <time
          className="reviews__time"
          dateTime="2020-10-15T13:58:19.494Z"
        >
          October 2020
        </time>
      </div>
    </li>
  </ul>,
]
`;
